### PR TITLE
Add support to Print::printf for printing from flash

### PR DESCRIPTION
**Pull request by me-no-dev**

### CODE REVIEW

**Comment by me-no-dev** [comment]
@igrr is travis borked?

File: cores/esp8266/Print.cpp [hunk 1]
```diff
@@ -63,6 +63,53 @@ size_t Print::printf(const char *format, ...) {
     return len;
 }
 
+size_t Print::printf(const __FlashStringHelper * format, ...) {
```
Comment (middelink) [inline cores/esp8266/Print.cpp]: How about introducing Print:vprintf(PGM_P format, va_list argp) and make the other 2 use it?
```C++
size_t Print::printf(const __FlashStringHelper * format, ...) {
	va_list arg;
	va_start(arg, format);
	PGM_P fmt = reinterpret_cast<PGM_P>(format);
	size_t len = vprintf(fmt, arg);
	va_end(arg);
	return len;
```

Comment (me-no-dev) [inline cores/esp8266/Print.cpp]: removed that printf all together. Check the latest commit

Comment (middelink) [inline cores/esp8266/Print.cpp]: Good enough.

**Comment by hallard** [comment]
@me-no-dev 
thanks, I was waiting for this one for so long....

**Pull request merged**